rust/src/transformations/make_stable_expr/namespace_arr/mod.rs

```diff
@@ -0,0 +1,42 @@
+use polars::prelude::FunctionExpr;
+use polars_plan::dsl::Expr;
+
+use crate::core::{MetricSpace, Transformation};
+use crate::domains::{ExprDomain, OuterMetric, WildExprDomain};
+use crate::error::*;
+use crate::transformations::DatasetMetric;
+
+use super::StableExpr;
+
+/// Make a Transformation that returns an expression under the `arr` namespace.
+///
+/// # Arguments
+/// * `input_domain` - Expr domain
+/// * `input_metric` - The metric under which neighboring LazyFrames are compared
+/// * `expr` - The arr expression
+pub fn make_namespace_arr<M: OuterMetric>(
+    _input_domain: WildExprDomain,
+    _input_metric: M,
+    expr: Expr,
+) -> Fallible<Transformation<WildExprDomain, ExprDomain, M, M>>
+where
+    M::InnerMetric: DatasetMetric,
+    M::Distance: Clone,
+    (WildExprDomain, M): MetricSpace,
+    (ExprDomain, M): MetricSpace,
+    Expr: StableExpr<M, M>,
+{
+    let Expr::Function {
+        function: FunctionExpr::ArrayExpr(array_function),
+        ..
+    } = &expr
+    else {
+        return fallible!(MakeTransformation, "expected array expression");
+    };
+
+    fallible!(
+        MakeTransformation,
+        "Expr is not recognized at this time: {:?}. Waiting for: https://github.com/pola-rs/polars/pull/20421", 
```

Comment:
Will it be a problem if that doesn't get addressed? If the maintainers aren't interested, would you take a different approach?

Would it be worth adding an expected-failure test that exercises this branch? I'm not sure exactly what it is that is blocked.

Reply:
If it doesn't get addressed then I'll have to pick apart the debug stringification. :/